overthebox/files/bin/otb-action-speedtest

```diff
@@ -43,8 +43,22 @@ speedtest() {
 			   | jq "(.end.sum_received.bits_per_second // 0) / 1000 | floor"
 	}
 
-	upload="$(do_iperf)" || return
-	download="$(do_iperf -R)" || return
+	# loop five times max to handle speedtest timeout when determining upload speed
+	for i in $(seq 1 1 5); do
+		upload="$(do_iperf)" || return
+		if [ $upload -ne 0 ]; then
```

Comment:
Could you add double quote to prevent globbing and word splitting ?
```
if [ "$upload" -ne 0 ]; then
```

Reply:
Yes, done.